dirty_cat/test/test_minhash_encoder.py

```diff
@@ -124,6 +126,21 @@ def test_missing_values(input_type, missing, hashing):
     return
 
 
+def test_cache_overflow():
+    # Regression test for cache overflow resulting in -1s in encoding
+    def get_random_string(length):
+        letters = ascii_lowercase
+        result_str = ''.join(random.choice(letters) for i in range(length))
+        return result_str
+
+    encoder = MinHashEncoder(n_components=3)
+    capacity = encoder.capacity
+    raw_data = [get_random_string(10) for x in range(capacity)]
```

Comment:
Should it be "capacity", or "capacity + 1"? A simple way of figuring this would be to see if this code fails with the old version of dirty_cat, where there was a bug.

Reply:
On release 0.0.7, I confirm using ``capacity + 1`` returns some ``-1``s.

![image](https://user-images.githubusercontent.com/33222654/98121184-458e5d00-1eaf-11eb-804c-fc48323f1c0d.png)

The same assertions launched from master returns no ``-1``s.

![image](https://user-images.githubusercontent.com/33222654/98121893-30fe9480-1eb0-11eb-9495-cfaa51141393.png)

So I assume we should add this second assertion (with ``== 0`` instead of ``== 1``).

Reply:
Good catch, thanks. Had tested on old version, but missed in the test. Should be fixed now.